### PR TITLE
Removes signature file adjacency check in TransparentCompiler

### LINKED ISSUE #17082: Transparent Compiler: Fails when signature file is not exactly next to source file
Please provide a succinct description of the issue.

**Repro steps**

Provide the steps required to reproduce the problem:

1. Create a project with multiple files that don't depend on each other
2. Use a wildcard to add all the files. see [fsautocomplete](https://github.com/ionide/FsAutoComplete/blob/ab6a49d7e423128884ca38b17c99cb520c039a69/src/FsAutoComplete/FsAutoComplete.fsproj#L37-L38)

**Expected behavior**

Typechecking completes

**Actual behavior**

Type checking fails with "Oops!!" exception

**Known workarounds**

Comment out https://github.com/dotnet/fsharp/blob/f68d7b60cafcac940754774ee34d8cf8e90ab754/src/Compiler/Service/TransparentCompiler.fs#L1418-L1422

**Related information**

Provide any related information (optional):

* Operating system
* .NET Runtime kind (.NET Core, .NET Framework, Mono)
* Editing Tools (e.g. Visual Studio Version, Visual Studio)

### PR DESCRIPTION
**Pull request by TheAngryByrd**
## Description

Comments out a check that can cause the TransparentCompiler to throw an exception if a signature file is next next to a source file.

Fixes #17082

## Checklist

- [ ] Test cases added
- [ ] Performance benchmarks added in case of performance changes
- [x] Release notes entry updated:
    > Please make sure to add an entry with short succinct description of the change as well as link to this pull request to the respective release notes file, if applicable.
    >
    > Release notes files:
    > - If anything under `src/Compiler` has been changed, please make sure to make an entry in `docs/release-notes/.FSharp.Compiler.Service/<version>.md`, where `<version>` is usually "highest" one, e.g. `42.8.200`
    > - If language feature was added (i.e. `LanguageFeatures.fsi` was changed), please add it to `docs/releae-notes/.Language/preview.md`
    > - If a change to `FSharp.Core` was made, please make sure to edit `docs/release-notes/.FSharp.Core/<version>.md` where version is "highest" one, e.g. `8.0.200`.

    > Information about the release notes entries format can be found in the [documentation](https://fsharp.github.io/fsharp-compiler-docs/release-notes/About.html).
    > Example:
    > * More inlines for Result module. ([PR #16106](https://github.com/dotnet/fsharp/pull/16106))
    > * Correctly handle assembly imports with public key token of 0 length. ([Issue #16359](https://github.com/dotnet/fsharp/issues/16359), [PR #16363](https://github.com/dotnet/fsharp/pull/16363))
    > *`while!` ([Language suggestion #1038](https://github.com/fsharp/fslang-suggestions/issues/1038), [PR #14238](https://github.com/dotnet/fsharp/pull/14238))

    > **If you believe that release notes are not necessary for this PR, please add `NO_RELEASE_NOTES` label to the pull request.**

### CODE REVIEW

**Comment by TheAngryByrd** [comment]
@dotnet-policy-service agree

**Review by vzarytovskii — APPROVED** [review]

**Review by edgarfgp — APPROVED** [review]

File: src/Compiler/Service/TransparentCompiler.fs [hunk 1]
```diff
@@ -1415,11 +1415,11 @@ type internal TransparentCompiler
                             if tcInfo.stateContainsNodes |> Set.contains fileNode then
                                 failwith $"Oops!"
 
-                            if
-                                tcInfo.stateContainsNodes
-                                |> Set.contains (NodeToTypeCheck.PhysicalFile(index + 1))
-                            then
-                                failwith $"Oops!!!"
+                            // if
```
Comment (psfinaki) [inline src/Compiler/Service/TransparentCompiler.fs:1418]: I'd say - either remove or comment :)

**Review by psfinaki — APPROVED** [review]
Thanks!

File: src/Compiler/Service/TransparentCompiler.fs (hunk 1 above)
Comment (TheAngryByrd) [reply src/Compiler/Service/TransparentCompiler.fs:1418]: Want me to remove the other [Oops??? in the PhysicalFile path](https://github.com/dotnet/fsharp/blob/f68d7b60cafcac940754774ee34d8cf8e90ab754/src/Compiler/Service/TransparentCompiler.fs#L1372-L1378) as well?

Comment (vzarytovskii) [reply src/Compiler/Service/TransparentCompiler.fs:1418]: Let's leave the comment, so we understand it was there in the first place, once we get to transparent compiler issues.

**Pull request merged**